daemons/controld/controld_based.c

```diff
@@ -46,6 +46,26 @@ do_cib_replaced(const char *event, xmlNode * msg)
     register_fsa_input(C_FSA_INTERNAL, I_ELECTION, NULL);
 }
 
+void
+disconnect_cib_manager(unsigned long long flags_to_clear)
+{
+    CRM_ASSERT(fsa_cib_conn != NULL);
+    
+    controld_clear_fsa_input_flags(flags_to_clear);
```

Comment:
I overlooked that the flag cleared was different in the two callers!

Calling lrm_state_destroy_all() and clearing R_LRM_CONNECTED are actually not part of disconnecting from the CIB -- instead, they are disconnecting from the executor (which used to be called the Local Resource Manager). Those two calls can stay separate in crmd_exit().

Clearing R_CIB_CONNECTED is the correct behavior in both places, so it can stay here. It was actually a minor bug that crmd_exit() didn't clear it before, but in practice it's unlikely to cause problems in the brief time it takes to exit. That sort of bug is actually a good example of why avoiding code duplication is a good idea :)